storage/innobase/dict/dict0load.cc

```diff
@@ -1800,6 +1852,208 @@ dict_load_virtual_low(
+
+/** Load a table added column definition from a SYS_COLUMNS_ADDED record to dict_table_t.
+@return	error message
+@retval	NULL on success */
+static
+const char*
+dict_load_column_added_low(
+	dict_table_t*	table,		/*!< in/out: table, could be NULL */
+	mem_heap_t*		heap,		/*!< in/out: memory heap for temporary storage */
+	const rec_t*	rec,		/*!< in: SYS_COLUMNS_ADDED record */
+	dict_index_t*   sys_index,
+	table_id_t*		table_id_out,
+	ulint*			pos_out,
+	char**			def_val_out,
+	ulint*			def_val_len_out)
+{
+	const byte*		field;
+	const byte*		def_val = NULL;
+	ulint			def_val_len = 0;
+	table_id_t		table_id;
+	ulint			len;
+	ulint			pos;
+
+	if (rec_get_deleted_flag(rec, 0)) {
+		return(dict_load_column_added_del);
+	}
+
+	if (rec_get_n_fields_old(rec) != DICT_NUM_FIELDS__SYS_COLUMNS_ADDED) {
+		return("wrong number of columns in SYS_COLUMNS_ADDED record");
+	}
+
+	field = rec_get_nth_field_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__TABLE_ID, &len);
+	if (len != 8) {
+err_len:
+		return("incorrect column length in SYS_COLUMNS_ADDED");
+	}
+
+	table_id = mach_read_from_8(field);
+	if (table && table->id != mach_read_from_8(field)) {
+		return("SYS_COLUMNS_ADDED.TABLE_ID mismatch");
+	}
+
+	field = rec_get_nth_field_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__POS, &len);
+	if (len != 4) {
+		goto err_len;
+	}
+
+	pos = mach_read_from_4(field);
+
+	rec_get_nth_field_offs_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__DB_TRX_ID, &len);
+	if (len != DATA_TRX_ID_LEN && len != UNIV_SQL_NULL) {
+		goto err_len;
+	}
+	rec_get_nth_field_offs_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__DB_ROLL_PTR, &len);
+	if (len != DATA_ROLL_PTR_LEN && len != UNIV_SQL_NULL) {
+		goto err_len;
+	}
+
+	if (UNIV_UNLIKELY(rec_offs_nth_extern_old(rec, 
+		DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE))) {
+		ulint       offsets_[REC_OFFS_NORMAL_SIZE];
+		ulint *     offsets = offsets_;
+
+		ut_a(rec);
+		rec_offs_init(offsets_);
+
+		offsets = rec_get_offsets(rec,sys_index,offsets,ULINT_UNDEFINED,&heap);
+
+		def_val = btr_rec_copy_externally_stored_field(
+				rec, offsets, 
+				dict_table_page_size(table),
+				DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE, &def_val_len, heap);
+		ut_ad(def_val_len);
+		ut_a(def_val);
+
+	} else {
+		field = rec_get_nth_field_old(
+			rec, DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE, &len);
+
+		if (len == UNIV_SQL_NULL) {
+			def_val = NULL;
+			def_val_len = UNIV_SQL_NULL;
+		} else {
+			def_val = field;
+			def_val_len = len;
+		}
+	}
+
+	if (table) {
+		dict_col_t* col = dict_table_get_nth_col(table, pos);
+
+		dict_col_set_added_column_default(col, def_val, def_val_len, table->heap);
+	}
+	
+	*table_id_out = table_id;
+	*pos_out = pos;
+	if(def_val_out && def_val_len_out) {
+		*def_val_out = def_val ? 
+							(char*)mem_heap_strdupl(heap, (char*)def_val, def_val_len) : 
+							(char*)def_val;
+		*def_val_len_out = def_val_len;
+	}
+	return(NULL);
+}
+
+static
+void
+dict_load_columns_added (
+	dict_table_t*	table,
+	mem_heap_t*		heap )
```

Comment:
Document the function and its parameters.
Return a status (such as `const char*`), so that the caller can perform error handling.
There should be no space before `(` in function declarations.
I would put all parameters in a single line when they fit, using spaces instead of TABs.